pkg/api/v1/helper/helpers.go

```diff
@@ -513,3 +519,70 @@ func StorageNodeAffinityToAlphaAnnotation(annotations map[string]string, affinit
 	annotations[v1.AlphaStorageNodeAffinityAnnotation] = string(json)
 	return nil
 }
+
+// ReorganizeTaints returns the updated set of taints, taking into account old taints that were not updated,
+// old taints that were updated, old taints that were deleted, and new taints.
+func ReorganizeTaints(node *v1.Node, overwrite bool, taintsToAdd []v1.Taint, taintsToRemove []v1.Taint) (string, []v1.Taint, error) {
```

Comment:
> Is there are policy for this such as when to update helper and when to update **util?**

I am not aware of anything like that.


So, there is no consensus on where these fns have to be put and both places seemed reasonable. Please look at https://github.com/kubernetes/kubernetes/issues/45060#issuecomment-299125377

The following are the reasons that I chose those respective places.
- /pkg/util/taints/taints.go already had the fns that are needed for taint operations like Set, string conversion etc which are kind of basic operations on taints(I saw them more like primitive operations. For ex, all the fn's here operate on string, []string etc).
- /pkg/v1/helper operates only on the taints on node and fn's in this package may need some fn's from util package.

Again, these are my views and I am fine with making changes.